trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/remote/RemoteSessionRole.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.remote;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public record RemoteSessionRole(String region, String roleArn, Optional<String> externalId)
```

Comment:
Should this be called `RealSessionRole`?

Reply:
Good question - I think this and `realCredentials` should be unified. But at the same time, I feel that `Remote` is a nicer name than `Real`.

What do people think of renaming the `real` credential to `remote`?

Reply:
heh - that was my next suggestion, `real` to `remote`. Maybe better for a different PR?

Reply:
@vagaerg I'll do a fast follow to rename the "real" stuff to "remote" everywhere.